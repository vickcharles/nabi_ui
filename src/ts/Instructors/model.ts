export interface ExperienceType {
  id?: number;
  employer: string;
  jobTitle: string;
  fromMonth: string;
  fromYear: string;
  toMonth: string;
  toYear: string;
  location: string;
}

export enum DegreeType {
  associate = 'Associate Degree',
  bachelors = 'Bachelor\'s Degree',
  graduate = 'Graduate Degreee',
  professional = 'Professional Degree',
  certification = 'Certification',
  other = 'Other'
}

export interface EducationType {
  id?: number;
  schoolName: string;
  graduationYear: string;
  degreeType: DegreeType;
  fieldOfStudy: string;
}

export enum PlaceForLessonsOptions {
  home = 'Student\'s home',
  studio = 'Instructor\'s studio',
  online = 'Online',
}

export interface PlaceForLessonsType {
  home: boolean;
  studio: boolean;
  online: boolean;
}

export enum LessonDuration {
  thirtyMins = '30 mins',
  fortyFiveMins = '45 mins',
  sixtyMins = '60 mins',
  ninetyMins = '90 mins'
}

export interface RatesType {
  thirtyMinsRate: number;
  fortyFiveMinsRate: number;
  sixtyMinsRate: number;
  ninetyMinsRate: number;
}

export enum SkillLevel {
  beginner = 'Beginner',
  intermediate = 'Intermediate',
  advanced = 'Advanced'
}

export interface InstrumentsType {
  instrument: string;
  skillLevel: SkillLevel;
}

export interface InstructorType {
  userId: string;
  bio?: string;
  instruments?: InstrumentsType[];
  rates?: RatesType;
  placeForLessons?: PlaceForLessonsType;
  studioAddress?: string;
  education?: EducationType[];
  experience?: ExperienceType[];
}

export type IState = InstructorType[];
