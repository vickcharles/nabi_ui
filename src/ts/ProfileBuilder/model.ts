export interface InstructorState {
  userId: string;
  bio?: string;
}

export type IState = InstructorState[];
  