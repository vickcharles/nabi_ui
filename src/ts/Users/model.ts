export enum HearAboutUs {
  Google = 'google',
  Facebook = 'facebook',
  Craigslist = 'craigslist',
  PrintAds = 'printAds',
  WordOfMouth = 'wordOfMouth',
  CurrentNabiUser = 'currentNabiUser',
  Other= 'other'
}

export interface HearAboutUsInfo {
  value: HearAboutUs;
  label: string;
}

export const hearAboutUsInfo = {
  Google: <HearAboutUsInfo> {
    value: HearAboutUs.Google,
    label: 'Google'
  },
  Facebook: <HearAboutUsInfo> {
    value: HearAboutUs.Facebook,
    label: 'Facebook'
  },
  Craigslist: <HearAboutUsInfo> {
    value: HearAboutUs.Craigslist,
    label: 'Craigslist'
  },
  PrintAds: <HearAboutUsInfo> {
    value: HearAboutUs.PrintAds,
    label: 'Print Ads'
  },
  WordOfMouth: <HearAboutUsInfo> {
    value: HearAboutUs.WordOfMouth,
    label: 'Word Of Mouth'
  },
  CurrentNabiUser: <HearAboutUsInfo> {
    value: HearAboutUs.CurrentNabiUser,
    label: 'Current Nabi User'
  },
  Other: <HearAboutUsInfo> {
    value: HearAboutUs.Other,
    label: 'Other'
  }
};

export enum Role {
  student = 'student',
  instructor = 'instructor',
  parent = 'parent'
}

export interface UserState {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  zipCode: string;
  role: Role;
  hearAboutUs: string;
  avatar?: string;
  displayName: string;
}

export type IState = UserState[];
