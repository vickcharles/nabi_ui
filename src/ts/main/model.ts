/**
 * State for firing redirect
 * @interface RedirectState
 */
export interface RedirectState {
  fireRedirect: boolean;
}
