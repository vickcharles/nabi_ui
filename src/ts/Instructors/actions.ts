import { Action } from 'redux';
import { InstructorActions } from './constants/ActionTypes';
import { InstructorState } from './model';

interface CreateInstructor extends Action {
  instructor: InstructorState;
}

interface UpdateInstructor extends Action {
  instructor: InstructorState;
}

/**
 * Action to create a new instructor
 * @param {instructor} instructor - The new instructor
 */
export function createInstructor(instructor: InstructorState): CreateInstructor {
  return {
    instructor,
    type: InstructorActions.CREATE_INSTUCTOR
  };
}

/**
 * Action to update instructor
 * @param {instructor} instructor - The updated instructor's state
 */
export function updateInstructor(instructor: InstructorState): UpdateInstructor {
  return {
    instructor,
    type: InstructorActions.UPDATE_INSTRUCTOR
  };
}
