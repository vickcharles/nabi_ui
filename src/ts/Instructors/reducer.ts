import { AnyAction } from 'redux';
import { InstructorType, IState } from './model';
import { InstructorActions } from './constants/ActionTypes';

const initialState: InstructorType[] = [];

export default function instructorsReducer(state: IState = initialState, action: AnyAction): IState {
  switch (action.type) {
    case InstructorActions.CREATE_INSTUCTOR:
      return [...state, action.instructor];

    case InstructorActions.UPDATE_INSTRUCTOR:
    return <IState> state.map(instructor =>
      instructor.userId === action.instructor.userId
        ? { 
            ...instructor, 
            ...action.instructor
        }
        : instructor
    );

    default:
      return state;
  }
}
