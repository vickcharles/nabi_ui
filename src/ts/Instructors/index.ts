export { default as ProfileBuilder } from './components/ProfileBuilder/ProfileBuilder';
export * from './actions';
import * as Instructor from './model';
export { Instructor };
import reducer from './reducer';
export default reducer;
