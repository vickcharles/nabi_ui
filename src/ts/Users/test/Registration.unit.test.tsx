import * as React from 'react';
import { Registration } from '../';
import { shallow } from 'enzyme';
import { UserState } from '../model';
import { InstructorType } from '../../Instructors/model';
import * as moment from 'moment';

const mockMoment: moment.Moment = moment('2018-06-20T15:19:25.000Z');

describe('Registration', () => {
  let wrapper: any;
  const createUser: (user: UserState) => {} = jest.fn();
  const createInstructor: (instructor: InstructorType) => {} = jest.fn();
  const searchZipCode: (user: UserState) => {} = jest.fn();
  const displayAgeDisclaimer = jest.spyOn(Registration.prototype, 'displayAgeDisclaimer');
  Date.now = jest.fn(() => 1487076708000);

  beforeEach(() => {
    wrapper = shallow(
      <Registration
        createUser={createUser}
        createInstructor={createInstructor}
        searchZipCode={searchZipCode}
      />
    );
  });

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Method handleChange()', () => {
    function test_handleChange(name: string, value: string): void {
      describe('When the event contains ' + name + ' for name and ' + value + ' for value', () => {
        beforeEach(() => {
          const event = {
            target: {
              name: name,
              value: value
            }
          };
          wrapper.instance().handleChange(event);
        });

        it('Sets the state\'s ' + name + ' to ' + value, () => {
          expect(wrapper.state(name)).toBe(value);
        });
      });
    }

    test_handleChange('firstName', 'Foo');
    test_handleChange('lastName', 'Bar');
    test_handleChange('email', 'Foobar');
    test_handleChange('pass', 'Elit');
    test_handleChange('zipcode', 'Quz');
    test_handleChange('email', 'Ore');
  });

  describe('Method handleBirthdayChange()', () => {
    beforeEach(() => {
      wrapper.instance().handleBirthdayChange(mockMoment);
    });

    it('Sets the birthday state', () => {
      expect(wrapper.state('birthday')).toEqual(mockMoment);
    });
  });

  describe('Method displayAgeDisclaimer()', () => {
    describe('When the user is 17 years old or younger', () => {
      beforeEach(() => {
        wrapper.setState({birthday: mockMoment});
        wrapper.instance().displayAgeDisclaimer();
      });

      it('Displays age disclaimer', () => {
        expect(wrapper.state('openModal')).toBeTruthy();
      });
    });

    describe('When the user is 18 years old or older', () => {
      const mockMomentYoung: moment.Moment = moment('1990-06-20T15:19:25.000Z');
      beforeEach(() => {
        wrapper.setState({birthday: mockMomentYoung});
        wrapper.instance().displayAgeDisclaimer();
      });

      it('Does not display age disclaimer', () => {
        expect(wrapper.state('openModal')).toBeFalsy();
      });
    });
  });

  describe('Method handleSubmit()', () => {
    beforeEach(() => {
      wrapper.instance().handleSubmit();
    });

    it('Calls displayAgeDisclaimer', () => {
      expect(displayAgeDisclaimer).toBeCalled();
    });

    it('Calls createUser', () => {
      expect(createUser).toBeCalled();
    });
  });
});
