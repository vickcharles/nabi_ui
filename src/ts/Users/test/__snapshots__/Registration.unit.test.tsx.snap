// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Registration
    createInstructor={[MockFunction]}
    createUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PageTitle
          pageTitle="REGISTRATION"
        />,
        <div
          className="nabi-background-white nabi-section"
        >
          <RegistrationForm
            handleChange={[Function]}
            handleSubmit={[Function]}
            hearAboutUs=""
            selectedRole="student"
          />
        </div>,
        false,
      ],
      "className": "nabi-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "pageTitle": "REGISTRATION",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <RegistrationForm
            handleChange={[Function]}
            handleSubmit={[Function]}
            hearAboutUs=""
            selectedRole="student"
          />,
          "className": "nabi-background-white nabi-section",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleChange": [Function],
            "handleSubmit": [Function],
            "hearAboutUs": "",
            "selectedRole": "student",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PageTitle
            pageTitle="REGISTRATION"
          />,
          <div
            className="nabi-background-white nabi-section"
          >
            <RegistrationForm
              handleChange={[Function]}
              handleSubmit={[Function]}
              hearAboutUs=""
              selectedRole="student"
            />
          </div>,
          false,
        ],
        "className": "nabi-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "pageTitle": "REGISTRATION",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RegistrationForm
              handleChange={[Function]}
              handleSubmit={[Function]}
              hearAboutUs=""
              selectedRole="student"
            />,
            "className": "nabi-background-white nabi-section",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleChange": [Function],
              "handleSubmit": [Function],
              "hearAboutUs": "",
              "selectedRole": "student",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
